### PR TITLE
swig 4.2.0

### PR DESCRIPTION
**Pull request by shogo82148**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/swig@4.1.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  license "GPL-3.0-or-later"
+
+  keg_only :versioned_formula
+
+  depends_on "pcre2"
+
+  uses_from_macos "python" => :test
+
+  def install
+    system "./autogen.sh" if build.head?
```
Comment (chenrui333) [inline Formula/s/swig@4.1.rb:15]: not needed
```suggestion
```

File: Formula/s/swig@4.1.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+class SwigAT41 < Formula
+  desc "Generate scripting interfaces to C/C++ code"
+  homepage "https://www.swig.org/"
+  url "https://downloads.sourceforge.net/project/swig/swig/swig-4.1.1/swig-4.1.1.tar.gz"
```
Comment (chenrui333) [inline Formula/s/swig@4.1.rb:4]: Is 4.1.x release line going to be supported?

**Comment by chenrui333** [comment]
@shogo82148 also you need an alias for swig.rb for 4.2 change

File: Formula/s/swig@4.1.rb (hunk 2 above)
Comment (shogo82148) [reply Formula/s/swig@4.1.rb:4]: I am not familiar with the SWIG project, so I do not know.
I could not decide whether it was okay to remove the alias for swig@4.1, so I left it as it is.

Comment (cho-m) [reply Formula/s/swig@4.1.rb:4]: > I am not familiar with the SWIG project, so I do not know.

In that case just remove the `swig@4.1` formula as versioned formulae need to meet standards listed at: https://docs.brew.sh/Versions#acceptable-versioned-formulae

Comment (shogo82148) [reply Formula/s/swig@4.1.rb:4]: OK, I'll remove it

**Review by ZhongRuoyu — DISMISSED** [review]

**Review by bevanjkay — APPROVED** [review]

**Review by chenrui333 — COMMENTED** [review]
👍

**Pull request merged**